File: 백준/Bronze/2747. 피보나치 수/피보나치 수.py
import sys
input = sys.stdin.readline
n = int(input())
dpTable = [-1] * (n + 1)
dpTable[0] = 0
dpTable[1] = 1

for i in range(2, n + 1):
  dpTable[i] = dpTable[i - 1] + dpTable[i - 2]

print(dpTable[n])

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        int[] dpTable = new int[n + 1];
        dpTable[0] = 0;
        dpTable[1] = 1;
        
        for (int i = 2; i < dpTable.length; i++) {
            dpTable[i] = (dpTable[i - 1] + dpTable[i - 2]) % 1234567;
        }
        
        return dpTable[n] % 1234567;
    }
}